src/lex/mod.rs

```diff
@@ -848,7 +849,9 @@ impl Iterator for Lexer {
                     }
                 }
                 x => {
-                    return Some(Err(self.span(span_start).with(LexError::UnknownToken(x))));
+                    return Some(Err(self
+                        .span(span_start)
+                        .with(LexError::UnknownToken(x as char))));
```

Comment:
I was imagining the `as char` cast in the Display impl ... but I guess it doesn't really matter.

Reply:
I think the errors should be as transparent as possible